url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/episodes/managers/MediaTombManager.java

```diff
@@ -0,0 +1,232 @@
+package stv6.episodes.managers;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.xerces.parsers.DOMParser;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import stv6.Profile;
+import stv6.STHandlerManager;
+import stv6.episodes.BasicEpisode;
+import stv6.episodes.EpisodeManager;
+import stv6.handlers.util.StaticMessageHandler;
+
+public class MediaTombManager extends AbstractManager implements EpisodeManager {
+
+	private static final int UPDATE_WAIT = 2500;
+	
+	private static final List<Path> basePaths = new LinkedList<Path>();
+	static {
+		basePaths.add(new Path("1", ""));
+	}
+	
+	private final String broadIp, trProfile;
+	private final int mtPort;
+	private final boolean trCheck;
+	private final List<String> trExtensions;
+	
+	private String sid;
+	
+	public MediaTombManager(String broadcastingIp, int mtport, String trProf, 
+			boolean trCheck, List<String> trExtensions) {
+		this.broadIp = broadcastingIp;
+		this.mtPort = mtport;
+		this.trProfile = trProf;
+		this.trCheck = trCheck;
+		this.trExtensions = trExtensions;
+	}
+
+	private boolean doSleep(int millis) {
+		try {
+			Thread.sleep(millis);
+		} catch (InterruptedException e) {}
+		
+		return true;
+	}
+	
+	@Override
+	protected List<Path> getBasePaths() {
+		return basePaths;
+	}
+
+	@Override
+	protected ArrayList<BasicEpisode> getEpisodesFor(Path path) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.